lib/matplotlib/axes/_axes.py

```diff
@@ -6273,28 +6273,23 @@ def _normalize_input(inp, ename='input'):
             totwidth = np.diff(bins)
 
             if rwidth is not None:
-                dr = min(1.0, max(0.0, rwidth))
+                dr = np.clip(rwidth, 0, 1)
             elif (len(n) > 1 and
                   ((not stacked) or rcParams['_internal.classic_mode'])):
                 dr = 0.8
             else:
                 dr = 1.0
 
             if histtype == 'bar' and not stacked:
-                width = dr*totwidth/nx
+                width = dr * totwidth / nx
                 dw = width
-
-                if nx > 1:
-                    boffset = -0.5*dr*totwidth*(1.0-1.0/nx)
-                else:
-                    boffset = 0.0
-                stacked = False
+                boffset = -0.5 * dr * totwidth * (1 - 1 / nx)
```

Comment:
It seems to me that due to l. 6176-6181 (and above), the case `nx == 0` should never occur: the case where the input is empty is preprocessed to have a length of 1.
```python
# Massage 'x' for processing.
if input_empty:
    x = np.array([[]])
else:
    x = _normalize_input(x, 'x')
nx = len(x)  # number of datasets
```